core/src/main/java/com/arcrobotics/ftclib/util/FTCLibOpMode.java

```diff
@@ -0,0 +1,45 @@
+package com.arcrobotics.ftclib.util;
+
+import com.arcrobotics.ftclib.gamepad.GamepadEx;
+import com.qualcomm.hardware.lynx.LynxModule;
+import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode;
+import com.qualcomm.robotcore.eventloop.opmode.OpMode;
+
+import java.util.List;
+
+public abstract class FTCLibOpMode extends LinearOpMode {
+
+    protected GamepadEx driverOp, toolOp;
+
+    @Override
+    public final void runOpMode() throws InterruptedException {
+        driverOp = new GamepadEx(gamepad1);
+        toolOp = new GamepadEx(gamepad2);
+
+        initialize();
+
+        List<LynxModule> hubs = hardwareMap.getAll(LynxModule.class);
+        hubs.forEach(hub -> hub.setBulkCachingMode(LynxModule.BulkCachingMode.MANUAL));
+
+        waitForStart();
+
+        while (!isStopRequested() && opModeIsActive()) {
+            hubs.forEach(LynxModule::clearBulkCache);
+            run();
+            driverOp.readButtons();
+            toolOp.readButtons();
+            telemetry.update();
+        }
+    }
+
+    /**
+     * This method is like {@link OpMode#init()}
+     */
+    public abstract void initialize();
+
+    /**
+     * This method is like {@link OpMode#loop()}
+     */
+    public abstract void run();
```

Comment:
The name does not make it clear that this method is called over and over. It's also inconsistent with command based—in that sense it should be `execute`; that's still does not make it clear it loops, but at least make it consistent.